File: BiliRecommendUI/src/main/java/com/bilirecommendui/main/widget/holder/RecommendFooterViewFactory.java
package com.bilirecommendui.main.widget.holder;

import android.view.View;

import com.bilirecommendui.App;
import com.bilirecommendui.R;

/**
 * by y on 2016/9/19
 */
public class RecommendFooterViewFactory {

    public static View createView(int id) {
        View view = null;
        switch (id) {
            case 0:
                view = getInflate(R.layout.item_recommend_footer_recommend);
                break;
            case 1:
                view = getInflate(R.layout.item_recommend_footer_fanju);
                break;
            case 2:
                view = getInflate(R.layout.item_recommend_footer_refresh_more);
                break;
            case 3:
                view = getInflate(R.layout.item_recommend_footer_games);
                break;
        }
        return view;
    }


    private static View getInflate(int id) {
        return View.inflate(App.getContext(), id, null);
    }
}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/main/widget/holder/RecommendFooterViewFactory.kt
package com.bilibilirecommend.main.widget.holder

import android.text.TextUtils
import android.view.View

import com.bilibilirecommend.App
import com.bilibilirecommend.R
import com.bilibilirecommend.main.model.RecommendModel
import com.bilibilirecommend.network.Constant


object RecommendFooterViewFactory {

    fun createView(type: RecommendModel.ResultBean): View {
        return when (type.type) {
            Constant.TYPE_RECOMMEND -> getInflate(R.layout.item_recommend_footer_recommend)
            Constant.TYPE_BANGUMI -> getInflate(R.layout.item_recommend_footer_fanju)
            Constant.TYPE_REGION -> {
                if (TextUtils.equals(type.head.title, "游戏区")) {
                    getInflate(R.layout.item_recommend_footer_games)
                } else getInflate(R.layout.item_recommend_footer_refresh_more)
            }
            else -> getInflate(R.layout.item_recommend_footer_refresh_more)
        }
    }


    private fun getInflate(id: Int): View {
        return View.inflate(App.instance, id, null)
    }
}
